lib/torrent.js

```diff
@@ -1174,8 +1174,15 @@ Torrent.prototype._updateInterest = function () {
   self._amInterested = !!self._selections.length
 
   self.wires.forEach(function (wire) {
-    // TODO: only call wire.interested if the wire has at least one piece we need
-    if (self._amInterested) wire.interested()
+    var interested = false
+    for (var index = 0; index < self.pieces.length; ++index) {
+      if (self.pieces[index] && wire.peerPieces.get(index)) {
```

Comment:
Shouldn't this `!self.pieces[index]`? Indicating that we don't yet have that piece?

Reply:
`self.pieces[index]` is deleted when that piece is downloaded and verified, so `self.pieces[index]` would be truthy when the piece still needs to be downloaded. The actual piece data is in the store. `self.pieces[]` is used to facilitate downloading.

`!self.bitfield.get(index)` is clearer.